src/utils/promise.ts

```diff
@@ -0,0 +1,20 @@
+type AsyncFunc<T, R> = (item: T) => Promise<R>;
+
+/**
+ * applies asyncFunc to each item and returns a promise with the array of results
+ * @param items
+ * @param asyncFunc
+ * @param initialValue
+ */
+export function sequential<T, R>(
```

Comment:
since in a few places you're calling `sequential` with `R = number`, you could write a `sequentialSum` in terms of `sequential`, that accept `asyncFunc: (item: T) => Promise<number>` and returns the sum of the values.